From 3b05904c1859d3c13f4ef47bd0d614ddfd4368d2 Mon Sep 17 00:00:00 2001
From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
Date: Mon, 15 Jul 2024 15:26:24 +0530
Subject: [PATCH 1/4] Avoid reading the Python include directory headers

This patch disables reading the Python include directory headers for the SWIG
codegen to identify the major version of Python. Instead, it directly sets the
Python version to 3 to avoid bothering with patchlevel.h, a file that usually
contains the version information for Python.

---
 CMakeLists.txt | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/swig/python/CMakeLists.txt b/swig/python/CMakeLists.txt
index 8f96af68..b588d685 100644
--- a/swig/python/CMakeLists.txt
+++ b/swig/python/CMakeLists.txt
@@ -28,10 +28,7 @@ if(APPLE)
 endif()

 function(CasadiGetPythonVersionMajor VERSION_MAJOR_RESULT INCLUDE_PATH)
-  set(MATCH_STRING "#define[ \t]+PY_MAJOR_VERSION[ \t]+([0-9]+)")
-  file(STRINGS "${INCLUDE_PATH}/patchlevel.h" PY_MAJOR_VERSION REGEX ${MATCH_STRING})
-  string(REGEX REPLACE "${MATCH_STRING}" "\\1" PYTHON_VERSION_MAJOR "${PY_MAJOR_VERSION}")
-  set(${VERSION_MAJOR_RESULT} ${PYTHON_VERSION_MAJOR} PARENT_SCOPE)
+  set(${VERSION_MAJOR_RESULT} 3 PARENT_SCOPE)
 endfunction()

 if(SWIG_EXPORT)
--
2.39.3 (Apple Git-146)
